localstorage.js

```diff
@@ -0,0 +1,34 @@
+const usernameInput = document.getElementById('fname');
+const emailInput = document.getElementById('email');
+const textareaInput = document.getElementById('textarea');
+
+// check local storage for user
+
+const userInfo = JSON.parse(localStorage.getItem('userInfo'));
+if (userInfo) {
+  usernameInput.value = userInfo.username;
+  emailInput.value = userInfo.email;
+  textareaInput.value = userInfo.textarea;
+}
+
+document.getElementById('form').addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  // get values from nput
+  const username = usernameInput.value.trim();
+  const email = emailInput.value.trim();
+  const textarea = textareaInput.value.trim();
+
+  if (!username || !email || !textarea) {
+    return;
+  }
+
+  // saving inputs
+  const userInfo = {
+    username,
+    email,
+    textarea,
+  };
+
```

Comment:
- [ ] Good job so far 👏  As an improvement to what you have done, It is important to note that the data are saved in the local storage every time the input value changes. That means the event that functions have to depend on is `OnChange` not `OnSubmit` as you have done. But you are almost there. I can see that data save when I click Get in touch button and retrieved back when I reload the page. So please work on the suggested improvement to complete it 😃 
![image](https://user-images.githubusercontent.com/90222110/209343157-8603c613-e601-4067-99d2-d6e2d50aaade.png)